### ISSUE TITLE
False positive with gas-struct-packing

### ISSUE THREAD

**Issue by RyanRHall** [protofire/solhint#597]
I think there is a false positive with the `gas-struct-packing` rule when using contract types. Here is the MRE:

```solidity
pragma solidity 0.8.24;

contract Bar {}

contract Foo {
  // this triggers gas-struct-packing
  struct MyStruct1 {
    uint96 a;
    Bar bar;
    address myAddr;
  }

  // this is fine
  struct MyStruct2 {
    uint96 a;
    address myAddr;
    Bar bar;
  }
}
```

`MyStruct1` and `MyStruct2` are both "tightly packed", but for some reason the linter doesn't like the first struct. I tested this against solhint v5.0.3

**Comment by jhweintraub** [comment]
I found another false positive also when using multiple addresses, since you can't pack 2 addresses into the same slot
```
struct ExampleStruct {
  address addr1; // 1 slot
  address addr2; // 1 slot
  address addr3; // 1 slot
}
```

**Comment by dbale-altoros** [comment]
@RyanRHall regarding your issue
As Bar is a contract , solhint cannot know slot space... If you look at the ast tree... Bar is a `UserDefinedTypeName`
Which by default is considered to take a whole new slot
That is why the first one fails and the second one, which is at the end, does not

**Comment by dbale-altoros** [comment]
@jhweintraub this is a good one... i need to check the code... thanks a lot for reporting!
